libs/hydrateAndFilter.js

```diff
@@ -72,29 +90,41 @@ const hydrateAndFilter = (entries, doctype, options = {}) => {
       const key = createHash(entry)
       if (store[key]) {
         entry._id = store[key]._id
+        entry._rev = store[key]._rev
       }
     })
     return entries
   }
 
-  const filterEntries = store => () => {
-    // filter out existing items
-    return bluebird.filter(entries, entry => {
-      return !store[createHash(entry)]
-    })
+  const defaultShouldSave = () => true
+  const defaultShouldUpdate = existing => false
+
+  const filterEntries = store => async () => {
+    // Filter out items according to shouldSave / shouldUpdate.
+    // Both can be passed as option or can be part of the entry.
+    return uniqBy(await bluebird.filter(entries, entry => {
+      const shouldSave = entry.shouldSave || options.shouldSave || defaultShouldSave
+      const shouldUpdate = entry.shouldUpdate || options.shouldUpdate || defaultShouldUpdate
```

Comment:
I find it weird that the filter function needs to know if an entry needs to be updated or created and this, again, feels  redundant with the createOrUpdate function. But maybe this is createOrUpdate which is obsolete.
But I this has more meaning in the `Model` vision we can see in your unit tests.

Why not define a `customFilter` option which takes a function and let the connector developer define it's own filter ? And maybe our own `Document` implementation would define these custom filters?